### ISSUE TITLE
Add metadata into the pipeline

### ISSUE THREAD

**Issue by ghost** [square/picasso#639]
Android L has a new feature that will extract common colors out of an image. Palette.generate(bitmap) This should be run on a background thread. I think it would be cool and efficient to provide a callback after the image has been loaded into memory(from disk or net) on the SAME thread so we can do more processing.  I don't think transitions are the correct place for this but I could be wrong and they also don't rerun transition on loading from disk which i would need it to do. I need to call that function and add the results to an internal cache for fast retrieval for activity animations.

I know i could use the Target class and kick off a new task to compute this but we already have thread that had to deal with the bitmap. Just seems inefficient to spin these up and I feel like this is going to need to be done a lot in the near future.

Also this is more of a request. I have no idea if this is the correct place for this.

Thanks
Joe

**Comment by dnkoutso** [comment]
Yes this will be done using generic means of metadata for requests.
